packages/ubuntu_init/lib/src/init_wizard.dart

```diff
@@ -112,11 +114,15 @@ class InitWizard extends ConsumerWidget {
           builder: (_) => const PrivacyPage(),
           onLoad: (_) => PrivacyPage.load(ref),
         ),
-        InitRoutes.store: WizardRoute(
-          builder: (_) => const StorePage(),
-          onLoad: (_) => StorePage.load(ref),
```

Comment:
Seems like you have accidentally removed the `StorePage` here when adding the new `LaunchSessionPage` - please restore it :)

Reply:
The problem is that there can't be two end pages: only the last one will show the "Done" button, while the other will show the "Next" button, and in the ubuntu-core-desktop-init program the StorePage isn't used.

But since this is common for all the wizards... how can we solve this?

Reply:
I didn't test it, but you should be able to solve this by adding `userData: const WizardRouteData(hasNext: false)` to both `InitRoutes.store` and `InitRoutes.launchsession`.

Reply:
I'll test it. Thanks!

Reply:
I tested, and it only shows the page that is placed before in the code. If I put first the code for the store page and in the ubuntu-provision.conf file I put the launchsession page, it just exits before showing the launch session page (but the store one works fine). If I put first the code for the launch session page and in the ubuntu-provision.conf file I put the store page, when I press "Next" in the previous page, it shows nothing and calls the "onNext" code of the launch session page... (!?)

Reply:
It seems like the "onNext" code is called no matter if the page was shown or not...

Reply:
A quick&dirty solution would be to put a public flag in the pages that is set to FALSE by default, and to TRUE if the page has been shown, and only run the specific code in the "onNext" callback if the flag is TRUE... What do you think?

Reply:
Had a quick look at this. You're right, it seems like the explicitly provided `onNext` is called even when the page is not included. I don't have the time right now to think about how to properly change this in `wizard_router`.
What we could do as a fix is check again in `InitRoute.store`'s `onNext` whether the route is actually included and then act accordingly, e.g.

```dart
InitRoutes.store: WizardRoute(
  builder: (_) => const StorePage(),
  onLoad: (_) => StorePage.load(ref),
  userData: const WizardRouteData(hasNext: false),
  onNext: (settings) async {
    if (_hasRoute(settings.name!) &&
        ref.read(initModelProvider).hasRoute(settings.name!)) {
      final window = YaruWindow.of(context);
      await _onDone?.call();
      await window.close();
      return InitRoutes.initial;
    }
    return InitRoutes.launchsession;
  },
),
InitRoutes.launchsession: WizardRoute(
  builder: (_) => const LaunchSessionPage(),
  onLoad: (_) => LaunchSessionPage.load(ref),
  userData: const WizardRouteData(hasNext: false),
  onNext: (_) async {
   ...
  },
),
```

I'd have to check if this as any unwanted implications, though

Reply:
I'll check this and other solutions...

Reply:
I found a simple way:

```
InitRoutes.launchsession: () {
          bool shownPage = false;
          return WizardRoute(
            userData: const WizardRouteData(hasNext: false),
            builder: (_) => const LaunchSessionPage(),
            onLoad: (_) {
              shownPage = true;
              return LaunchSessionPage.load(ref);
            },
            onNext: (_) async {
              if (shownPage) {
                final window = YaruWindow.of(context);
                GdmService gdmService = getService<GdmService>();
                await gdmService.openNewSession();
                await _onDone?.call();
                await window.close();
                return InitRoutes.initial;
              } else {
                return null;
              }
            }
          );
        }(),
```